bn-apps/memberships-management/membership-service/src/main/kotlin/com/r3/businessnetworks/membership/flows/bno/SelfIssueMembershipFlow.kt

```diff
@@ -2,60 +2,100 @@ package com.r3.businessnetworks.membership.flows.bno
 
 import co.paralleluniverse.fibers.Suspendable
 import com.r3.businessnetworks.membership.flows.bno.service.BNOConfigurationService
-import com.r3.businessnetworks.membership.flows.bno.service.DatabaseService
-import com.r3.businessnetworks.membership.flows.getAttachmentIdForGenericParam
-import com.r3.businessnetworks.membership.flows.isAttachmentRequired
-import com.r3.businessnetworks.membership.flows.member.RequestMembershipFlow
+import com.r3.businessnetworks.membership.flows.bno.support.BusinessNetworkOperatorFlowLogic
 import com.r3.businessnetworks.membership.flows.member.Utils.throwExceptionIfNotBNO
 import com.r3.businessnetworks.membership.states.MembershipContract
 import com.r3.businessnetworks.membership.states.MembershipState
 import com.r3.businessnetworks.membership.states.MembershipStatus
-import net.corda.core.contracts.Command
-import net.corda.core.flows.*
+import net.corda.core.contracts.StateAndRef
+import net.corda.core.flows.FinalityFlow
+import net.corda.core.flows.FlowLogic
+import net.corda.core.flows.InitiatingFlow
+import net.corda.core.flows.StartableByRPC
+import net.corda.core.identity.Party
 import net.corda.core.transactions.SignedTransaction
 import net.corda.core.transactions.TransactionBuilder
 import net.corda.core.utilities.ProgressTracker
-import java.time.Instant
 
 /**
- * Self Issue a Membership.  This can only be done by a BNO
+ * Self Issue a Membership.  This can only be done by a BNO,
+ * @param bnoMembership bno membership state to be activated
  */
+@Suppress("UNUSED_VARIABLE")
 @StartableByRPC
 @InitiatingFlow(version = 2)
-open class SelfIssueMembershipFlow(val metaData : Any) : FlowLogic<SignedTransaction>() {
+open class SelfIssueMembershipFlow(val bnoMembership: StateAndRef<MembershipState<Any>>) : FlowLogic<SignedTransaction>() {
```

Comment:
Hi Arif. As far as I understand from the class name, this flow is supposed to issue a membership onto the ledger, right? Why is it taking an *existing* membership as a parameter then?

Reply:
The parameter is just for the BNO to enter their Identity so I can use it to first check if they are in the list of whitelisted BNO's and then change the BNO status to Active member of the network

Reply:
Yes, but *before* activating a membership, you'd need to issue it first, right? How can BNO do that?

Reply:
From my understanding the BNO node is created from the MembershipState already and this workflow was supposed to just change the status parameter of the BNO node to active

Reply:
If the BNO is just a party I will need to create a membership state for it?

Reply:
Yes BNO needs to be issued with a membership state (just as any other participant on the network). The lifecycle of membership for BNO is exactly the same as for everyone else, i.e. Issue -> Activate -> Revoke. The difference is only that for BNO, both member and BNO fields would be the same.